third_party/move/move-compiler-v2/src/file_format_generator/function_generator.rs

```diff
@@ -710,11 +780,36 @@ impl<'env> FunctionContext<'env> {
 
     /// Returns true of the given temporary can/should be copied when it is loaded onto the stack.
     /// Currently, this is using the `Copy` ability, but in the future it may also use lifetime
-    /// analysis results to check whether the variable is still accessed.
+    /// pipeline results to check whether the variable is still accessed.
     pub fn is_copyable(&self, temp: TempIndex) -> bool {
         self.module
             .env
             .type_abilities(self.temp_type(temp), &self.type_parameters)
             .has_ability(FF::Ability::Copy)
     }
+
+    /// Determine whether the temporary is alive (used) in the reachable code after this point.
+    pub fn is_alive_after(&self, code_offset: FF::CodeOffset, temp: TempIndex) -> bool {
+        let LiveVarAnnotation(map) = self
```

Comment:
It is nearly zero cost since a reference to the inner map is returned. But you are right about the abstraction, in fact there was already a function `get_live_var_info_at` for the annotation type. Done.